src/django_mysql/models/aggregates.py

```diff
@@ -53,6 +54,15 @@ def as_sql(
         connection: BaseDatabaseWrapper,
         **extra_context: Any,
     ) -> tuple[str, tuple[Any, ...]]:
+        if self.filter:
+            extra_context["distinct"] = "DISTINCT " if self.distinct else ""
+            copy = self.copy()
+            copy.filter = None
+            source_expressions = copy.get_source_expressions()
+            condition = When(self.filter, then=source_expressions[0])
+            copy.set_source_expressions([Case(condition)] + source_expressions[1:])
+            return super(Aggregate, copy).as_sql(compiler, connection, **extra_context)
```

Comment:
I haven't looked into it, but it's a bit odd that you're skipping the existing implementation and calling `super()`. The implemetnation below covers the SQL oddities in `GROUP_CONCAT`, I think we'd still want to use it, just tweak adding the filter there.

Reply:
I'm not sure we need keep the existing pathway in the long term. It's probably better to let `Aggregate.as_sql()` perform the work based on the template.

If you look at the code for `Count`, you'll see there are 2 paths, one when filter is set, the other when it's not. I think we should have a similar structure here, but avoid manual handcrafting as much as possible because this is likely fragile.